src/frequenz/sdk/config/_util.py

```diff
@@ -0,0 +1,45 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Utilities to deal with configuration."""
+
+from collections.abc import Mapping
+from typing import Any, TypeVar, cast
+
+from marshmallow_dataclass import class_schema
+
+T = TypeVar("T")
+"""Type variable for configuration classes."""
+
+
+def load_config(
+    cls: type[T],
+    config: Mapping[str, Any],
+    /,
+    **marshmallow_load_kwargs: Any,
+) -> T:
+    """Load a configuration from a dictionary into an instance of a configuration class.
+
+    The configuration class is expected to be a [`dataclasses.dataclass`][], which is
+    used to create a [`marshmallow.Schema`][] schema to validate the configuration
+    dictionary.
+
+    To customize the schema derived from the configuration dataclass, you can use
+    [`marshmallow_dataclass.dataclass`][] to specify extra metadata.
+
+    Additional arguments can be passed to [`marshmallow.Schema.load`][] using keyword
+    arguments.
+
+    Args:
+        cls: The configuration class.
+        config: The configuration dictionary.
+        **marshmallow_load_kwargs: Additional arguments to be passed to
+            [`marshmallow.Schema.load`][].
+
+    Returns:
+        The loaded configuration as an instance of the configuration class.
```

Comment:
I understand the function is internal but IMO the documentation should include the `Raises` section as it is the only way the caller of the function can handle validation errors.

Reply:
I agree it could be a good addition, but the docs say very clearly that this is just a wrapper for `marshmallow.Schema.load()`, so I guess users can infer it from there. I can add it in a separate PR if you think it is worth it though.

Reply:
> I can add it in a separate PR if you think it is worth it though.

No, it's probably not worth it. Though I'd explicitly add it if the function is exposed to the user at some point

Reply:
It is already exposed via `frequenz.sdk.config` though :)

Reply:
oops, would you mind to add it in a different PR then? :angel: 

I have some other concern that I'd like to mention before adding any `Raise` docs.

```python
def test_none_settings() -> None:
    """Test loading None settings."""
    config: dict[str, Any] = {"key": "value"}

    text: str = "text"

    with pytest.raises(marshmallow.ValidationError) as exc_info:
        loaded_config = load_config(SimpleConfig, None)  # mypy complains incompatible type "None";

    with pytest.raises(marshmallow.ValidationError) as exc_info:
        loaded_config = load_config(
            SimpleConfig, config.get("unknown", None)
        )  # mypy is happpy with `None` when `get()` is used
```

Since `mypy` doesn’t raise any issues when get() is used, we can assume marshmallow will handle it. Marshmallow will raise `ValidationError` for the `None` case if default marshmallow parameters are used but that might be different when non-default parameters are set, for instance, `unknown=INCLUDE` will make marshmallow to raise a TypeError if None is passed as a `config`.

Reply:
Just for the record, we had a quick call and decided it would be impossible to document every pitfall we can have with marshmallow in this function, so we'll leave it like this. If we touch this code again, the docs can be made a bit more clear about this being just a wrapper over `marshmallow.Schema.load()`.